Fix in non-ANSI language, status board cannot be shown

Zipher04 opened the pull request:
The statusNameL may contain non-ANSI characters, which cannot be used as array index.
The status code needed for m_statusByColumns is directly calculated by PHP.

Zipher04 committed:
Update main.php

Add a new class "status_code_order". Which return the status code in order of the status defined in the plugin configuration.

Zipher04 committed:
Get status code directly from PHP

The statusNameL may contain non-ANSI characters, which cannot be used as array index.
The status code is get from PHP directly.

AuthenticEshkinKot commented:
Hi, @Zipher04 !
Thanks for your PR. I would like to merge it, but I have one question - why did you add "status_code_order"? As far as I can see, you don't use it.

Zipher04 committed:
Fix error

Fixed getStatusCodes_st() should get value from "status_code_order" from php page.

Zipher04 commented:
Hello, @AuthenticEshkinKot 
Sorry for the error. The "status_code_order" should be used by getStatusCodes_st(). Please see the fix above. Thank you!

AuthenticEshkinKot commented on Taskodrome/pages/main.php:
```diff
@@ -160,7 +160,25 @@ function write_bug_rows( $p_rows )
+    
+    foreach( $status_name as $t_name ) {
+        for ( $i = 0 ; $i < sizeof($t_status_array) ; $i++ ){
+            if ( 0 == strcmp($t_name, $t_status_array[$i][1]) ){
```
You're trying to compare status name from mantis ($t_status_array[$i][1], from config_get( 'status_enum_string' )), that is always english, and localized status name ($t_name, from $status_order), that may be on any other language. In such a case, this code compare will never return "true".

AuthenticEshkinKot replied:
@Zipher04 
Would you like to fix it?
